File: frontend/src/types/trend.ts
export interface Trend {
  id: number
  keyword: string
  score: number
  source: string
  category: string | null
  platform: string | null
  region: string
  date: string
}

export interface TrendAnalysis {
  summary: string
  recommendations: string[]
  topKeywords: Trend[]
}

export interface TrendAlert {
  id: number
  keyword: string
  threshold: number
  enabled: boolean
  createdAt: string | null
}

export interface CreateTrendAlertRequest {
  keyword: string
  threshold?: number
}

export interface UpdateTrendAlertRequest {
  keyword?: string
  threshold?: number
  enabled?: boolean
}

File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/trend/Trend.kt
package com.ongo.domain.trend

import java.time.LocalDate
import java.time.LocalDateTime

data class Trend(
    val id: Long? = null,
    val category: String? = null,
    val keyword: String,
    val score: Double = 0.0,
    val source: String,
    val platform: String? = null,
    val region: String = "KR",
    val date: LocalDate = LocalDate.now(),
    val metadata: String? = null,
    val createdAt: LocalDateTime? = null,
)
